File: pwd_takeinp.py
#!/usr/bin/env python
import os, sys

def splitall(path: str):
    allparts = []
    while 1:
        parts = os.path.split(path)
        if parts[0] == path:  # sentinel for absolute paths
            allparts.insert(0, parts[0])
            break
        elif parts[1] == path: # sentinel for relative paths
            allparts.insert(0, parts[1])
            break
        else:
            path = parts[0]
            allparts.insert(0, parts[1])
    return allparts

def colorize(inp: str) -> str:
    return "\033[38;2;102;217;239m" + inp + "\033[0;10m"


def main():
    pwd = sys.argv[1]

    dirs = splitall(pwd)
    dirs.pop(0)
    dirs[0] = '/' + dirs[0]

    ind = 0
    for path_part in dirs[:-1]:
        print(colorize(path_part))
        print(f"{(' ' * 4 * ind)}└── ", end='')
        ind += 1

    print(colorize(dirs[-1]))

if __name__ == '__main__':
    main()

File: pwd_takeinp.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> splitall(string& pathname) {
    string delim = "/";
    vector<string> res {};

    size_t pos = 0;
    string token;
    while ((pos = pathname.find(delim)) != string::npos) {
        token = pathname.substr(0, pos);
        res.push_back(token);
        pathname.erase(0, pos + delim.length());
    }
    res.push_back(pathname);
    return res;
}

string colorize(string inp) {
    return "\033[38;2;102;217;239m" + inp + "\033[0;10m";
}

int main(int argc, char* argv[]) {
    string pwd = argv[1];

    vector<string> dirs = splitall(pwd);
    dirs.erase(dirs.begin());

    int dirs_size = dirs.size();
    cout << colorize("/");
    for (int i = 0; i < dirs_size - 1; i++) {
        cout << colorize(dirs[i]) << endl;
        for (int j = 0; j < (4 * i); j++) {
            cout << ' ';
        }
        cout << "└── ";
    }

    cout << colorize(dirs[dirs_size - 1]) << endl;
}
